### PR TITLE
Add ParserTest

### PR DESCRIPTION
**Pull request by Zhiyuan-Amos**

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/ParserTest.java [hunk 1]
```diff
@@ -3,17 +3,79 @@
...
+        DeleteCommand command = (DeleteCommand) parser.parseCommand(DeleteCommand.COMMAND_WORD + " 1");
+        assertEquals(new DeleteCommand(INDEX_FIRST_PERSON), command);
+    }
+
+    @Test
+    public void parseCommand_edit() throws Exception {
+        String editedName = "foo";
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withName(editedName).build();
+        EditCommand command = (EditCommand) parser.parseCommand(
+                EditCommand.COMMAND_WORD + " 1 " + PREFIX_NAME + editedName);
```
Comment (yamgent) [inline src/test/java/seedu/address/logic/parser/ParserTest.java]: This can be more thorough

**Comment by CanIHasReview** [comment]
# v1

@Zhiyuan-Amos submitted v1 for review.

* [v1 1/1] [Create ParserTest](https://github.com/se-edu/addressbook-level4/pull/538/commits/7160ba93ca8079058e317221b268fe00c42cd564)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5a70b77b2a4679666e12e36fe92dccc4f53f5310...7160ba93ca8079058e317221b268fe00c42cd564))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/538/1/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/test/java/seedu/address/logic/parser/ParserTest.java [hunk 2]
```diff
@@ -3,17 +3,96 @@
...
+    public void parseCommand_delete() throws Exception {
+        DeleteCommand command = (DeleteCommand) parser.parseCommand(
+                DeleteCommand.COMMAND_WORD + " " + INDEX_FIRST_PERSON.getOneBased());
+        assertEquals(new DeleteCommand(INDEX_FIRST_PERSON), command);
+    }
+
+    @Test
+    public void parseCommand_edit() throws Exception {
+        Person person = new PersonBuilder().build();
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withName(DEFAULT_NAME)
```
Comment (yamgent) [inline src/test/java/seedu/address/logic/parser/ParserTest.java]: Can just use `new EditPersonDescriptorBuilder(person);`?

File: src/test/java/seedu/address/logic/parser/ParserTest.java [hunk 3]
```diff
@@ -3,17 +3,96 @@
...
+    @Test
+    public void parseCommand_exit() throws Exception {
+        assertTrue(parser.parseCommand(ExitCommand.COMMAND_WORD) instanceof ExitCommand);
+        assertTrue(parser.parseCommand(ExitCommand.COMMAND_WORD + " 3") instanceof ExitCommand);
+    }
+
+    @Test
+    public void parseCommand_find() throws Exception {
+        HashSet<String> keywords = new HashSet<>(Arrays.asList("foo", "bar", "baz"));
+        FindCommand command = (FindCommand) parser.parseCommand(FindCommand.COMMAND_WORD + " " + "foo bar baz");
```
Comment (yamgent) [inline src/test/java/seedu/address/logic/parser/ParserTest.java]: ```keywords.stream().collect(Collectors.joining(" "));```

**Comment by CanIHasReview** [comment]
# v2

@Zhiyuan-Amos submitted v2 for review.

* [v2 1/1] [Create ParserTest](https://github.com/se-edu/addressbook-level4/pull/538/commits/fd2cd25d6153a20284f0c1d8e88715ee8de318ec)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5a70b77b2a4679666e12e36fe92dccc4f53f5310...fd2cd25d6153a20284f0c1d8e88715ee8de318ec)) ([:chart_with_upwards_trend: Interdiff between v1 and v2](https://github.com/se-edu/addressbook-level4/compare/d34c9a928cd9e463bd46b37dc4daf3f105d2feb1~1...d34c9a928cd9e463bd46b37dc4daf3f105d2feb1))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/538/2/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

File: src/test/java/seedu/address/testutil/PersonUtil.java [hunk 4]
```diff
@@ -24,7 +24,7 @@ public static String getAddCommand(Person person) {
     /**
      * Returns the part of command string for the given {@code person}'s details.
      */
-    private static String getPersonDetails(Person person) {
+    public static String getPersonDetails(Person person) {
```
Comment (limmlingg) [inline src/test/java/seedu/address/testutil/PersonUtil.java]: Should be in different commit (or at least acknowledged in the commit message.)

**Comment by CanIHasReview** [comment]
# v3

@Zhiyuan-Amos submitted v3 for review.

* [v3 1/1] [Create ParserTest](https://github.com/se-edu/addressbook-level4/pull/538/commits/5e310af48ed44f9f2f56e747c5f6d863c169267e)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/5a70b77b2a4679666e12e36fe92dccc4f53f5310...5e310af48ed44f9f2f56e747c5f6d863c169267e)) ([:chart_with_upwards_trend: Interdiff between v2 and v3](https://github.com/se-edu/addressbook-level4/compare/862b89ae7cfb4f736da9662c59cacc80e753d282~1...862b89ae7cfb4f736da9662c59cacc80e753d282))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/538/3/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by damithc** [comment]
Commit message (and PR title) doesn't mach the diff?
Also, we usually say `Add XYZ` rather than `Create XYZ`

**Comment by Zhiyuan-Amos** [comment]
noted, will split it up

**Comment by CanIHasReview** [comment]
# v4

@Zhiyuan-Amos submitted v4 for review.

* [v4 1/2] [PersonUtil: Update getPersonDetails\(…\) visibility modifier to public](https://github.com/se-edu/addressbook-level4/pull/538/commits/1b07ea652e4a20cbdb8160ce8219b3b1848c009c)
* [v4 2/2] [Create ParserTest](https://github.com/se-edu/addressbook-level4/pull/538/commits/c1f2d19c4c83dc60dbac3049f1c232a783d2d2fe)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/89889b9a0bb1c75b945d41466adc4bec50f9a72e...c1f2d19c4c83dc60dbac3049f1c232a783d2d2fe)) ([:chart_with_upwards_trend: Interdiff between v3 and v4](https://github.com/se-edu/addressbook-level4/compare/96e8edd95c3931896b2411a268b84edd25decd30~1...96e8edd95c3931896b2411a268b84edd25decd30))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/538/4/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by CanIHasReview** [comment]
# v5

@Zhiyuan-Amos submitted v5 for review.

* [v5 1/2] [PersonUtil: Update getPersonDetails\(…\) visibility modifier to public](https://github.com/se-edu/addressbook-level4/pull/538/commits/1b07ea652e4a20cbdb8160ce8219b3b1848c009c)
* [v5 2/2] [Add ParserTest](https://github.com/se-edu/addressbook-level4/pull/538/commits/ad2a726c0897febc972784b0c7bda415041b970b)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/89889b9a0bb1c75b945d41466adc4bec50f9a72e...ad2a726c0897febc972784b0c7bda415041b970b)) ([:chart_with_upwards_trend: Interdiff between v4 and v5](https://github.com/se-edu/addressbook-level4/compare/0e8060eaee0b43f6b962af40ec8525da599e62ab~1...0e8060eaee0b43f6b962af40ec8525da599e62ab))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/538/5/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>

**Comment by Zhiyuan-Amos** [comment]
Apologies, please ignore v.4 as I did not change the commit message title from `Create XYZ` to `Add XYZ`.

**Comment by CanIHasReview** [comment]
# v6

@Zhiyuan-Amos submitted v6 for review.

* [v6 1/2] [PersonUtil: Update getPersonDetails\(\.\.\.\) visibility modifier to public](https://github.com/se-edu/addressbook-level4/pull/538/commits/8dc5b6e49b781994f7309a83740a08af55cd18f5)
* [v6 2/2] [Add ParserTest](https://github.com/se-edu/addressbook-level4/pull/538/commits/e03ce4f16937da11ff9269d447f66c955ecbc7b9)

([:books: Archive](https://github.com/se-edu/addressbook-level4/compare/89889b9a0bb1c75b945d41466adc4bec50f9a72e...e03ce4f16937da11ff9269d447f66c955ecbc7b9)) ([:chart_with_upwards_trend: Interdiff between v5 and v6](https://github.com/se-edu/addressbook-level4/compare/2ffa228db724bb36351c8e467bdb375c7274ba42~1...2ffa228db724bb36351c8e467bdb375c7274ba42))
<details>
<summary>Checkout this PR version locally</summary>

```sh
git fetch https://github.com/se-edu/addressbook-level4.git refs/pr/538/6/head:BRANCHNAME
```
where `BRANCHNAME` is the name of the local branch you wish to fetch this PR to.

</details>
